packages/peregrine/lib/talons/AuthModal/useAuthModal.js

```diff
@@ -67,6 +67,7 @@ export const useAuthModal = props => {
 
         // Also delete the Cart data from the apollo cache.
         await deleteCacheEntry(apolloClient, key => key.match(/^Cart/));
+        await deleteCacheEntry(apolloClient, key => key.match(/^\$Cart/));
```

Comment:
Maybe add a comment about the dual rule (Apollo caches weird when it can't compute a key).
```suggestion
        await deleteCacheEntry(apolloClient, key => key.match(/^\$?Cart/));
```

Reply:
This is an interesting point - we're explicit about which keys to delete, but by combining them in a single regex/function call that point is kind of obscured. Do you think its worth it to combine? I can add a comment above each call just to explain, at least.

Reply:
Using regex obscures what you're doing for most anyway, I don't see any problem with being more efficient. If you were using a string method I would agree, eg. `key.startsWith`.

Reply:
acf791049